avocado/core/settings.py

```diff
@@ -107,6 +107,8 @@ def convert_value_type(value, value_type):
             value_type = float
         elif value_type == 'list':
             value_type = list
+        elif value_type == 'path':
+            value_type = os.path.expanduser
```

Comment:
On the other hand it warns you that `TypeError` is not the only exception you might see.